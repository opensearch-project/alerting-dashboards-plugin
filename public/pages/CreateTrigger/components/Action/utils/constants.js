/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import Message from '../actions/index';

export const ActionsMap = {
  slack: {
    label: 'Slack notification',
    component: (props) => <Message {...props} />,
  },
  chime: {
    label: 'Amazon Chime notification',
    component: (props) => <Message {...props} />,
  },
  custom_webhook: {
    label: 'Custom webhook',
    component: (props) => <Message isSubjectDisabled {...props} />,
  },
  webhook: {
    label: 'Custom webhook',
    component: (props) => <Message isSubjectDisabled {...props} />,
  },
  email: {
    label: 'Email notification',
    component: (props) => <Message {...props} />,
  },
};
