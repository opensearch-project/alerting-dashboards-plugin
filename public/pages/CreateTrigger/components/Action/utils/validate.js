/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const validateDestination = (destinations, isFullText) => (value) => {
  if (!value)
    return isFullText ? 'Please select a channel or remove this notification.' : 'Required.';
  // In case existing destination doesn't exist in list , invalidate the field
  const destinationMatches = destinations.filter((destination) => destination.value === value);
  if (destinationMatches.length === 0) {
    return 'Required.';
  }
};
