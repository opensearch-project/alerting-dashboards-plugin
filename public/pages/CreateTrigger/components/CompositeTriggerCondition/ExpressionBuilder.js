import React, { useEffect, useState, useCallback } from 'react';
import {
  EuiFlexItem,
  EuiFlexGroup,
  EuiPopover,
  EuiComboBox,
  EuiButtonIcon,
  EuiExpression,
  EuiToolTip,
} from '@elastic/eui';
import * as _ from 'lodash';
import { FormikFormRow, FormikInputWrapper } from '../../../../components/FormControls';
import { getMonitors } from '../../../CreateMonitor/components/AssociateMonitors/AssociateMonitors';

export const conditionToExpressions = (condition = '', monitors) => {
  if (!condition.length) return [];

  const conditionMap = {
    '&&': 'AND',
    '||': 'OR',
    '!': 'NOT',
    '': '',
    '&& !': 'AND NOT',
    '|| !': 'OR NOT',
  };
  const queryToExpressionRegex = new RegExp(
    /(!|| && || \|\| || && \!|| \|\| \!)?(monitor\[id=(.*?)\])/,
    'gm'
  );
  const matcher = condition.matchAll(queryToExpressionRegex);
  let match;
  let expressions = [];
  let counter = 0;
  while ((match = matcher.next().value)) {
    if (counter && !match[1]) return []; // Didn't find condition after the first match

    const monitorId = match[3]?.trim(); // match [3] is the monitor_id
    const monitor = monitors.filter((mon) => mon.monitor_id === monitorId);
    expressions.push({
      description: conditionMap[match[1]?.trim()] || '', // match [1] is the description/condition
      isOpen: false,
      monitor_name: monitor[0]?.monitor_name,
      monitor_id: monitorId,
    });

    counter++;
  }

  return expressions;
};

const ExpressionBuilder = ({
  formikFieldPath = '',
  formikFieldName,
  values,
  touched,
  httpClient,
  edit,
  triggerIndex,
}) => {
  const formikFullFieldName = `${formikFieldPath}${formikFieldName}`;
  const formikFullFieldValue = _.replace(`${formikFullFieldName}_value`, /[.\[\]]/gm, '_');
  const expressionNamePrefix = `expressionQueries_${triggerIndex}`;

  const DEFAULT_CONDITION = 'AND';
  const DEFAULT_NAME = 'Select delegate monitor';
  const DEFAULT_EXPRESSION = {
    description: '',
    isOpen: false,
    monitor_id: '',
    monitor_name: DEFAULT_NAME,
  };
  const DEFAULT_NEXT_EXPRESSION = {
    ...DEFAULT_EXPRESSION,
    description: DEFAULT_CONDITION,
  };
  const FIRST_EXPRESSION_CONDITIONS_MAP = [
    { description: '', label: '' },
    { description: 'NOT', label: 'NOT' },
  ];
  const EXPRESSION_CONDITIONS_MAP = [
    { description: 'AND', label: 'AND' },
    { description: 'OR', label: 'OR' },
    { description: 'AND NOT', label: 'AND NOT' },
    { description: 'OR NOT', label: 'OR NOT' },
  ];

  const [usedExpressions, setUsedExpressions] = useState([DEFAULT_EXPRESSION]);
  const [options, setOptions] = useState([]);
  const triggerConditions = _.get(values, formikFullFieldName, '');

  useEffect(() => {
    // initializing formik because these are generic fields and formik won't pick them up until fields is updated
    !_.get(touched, formikFullFieldValue) && _.set(touched, formikFullFieldValue, false);
    !_.get(values, formikFullFieldValue) && _.set(values, formikFullFieldValue, '');

    const monitors = _.get(values, 'monitorOptions', []);
    if (monitors.length) {
      setInitialValues(monitors);
    } else {
      getMonitors(httpClient).then((monitors) => {
        _.set(values, 'monitorOptions', monitors);
        setInitialValues(monitors);
      });
    }
  }, [values.associatedMonitors?.sequence?.delegates, triggerConditions]);

  const setInitialValues = (monitors) => {
    const monitorOptions = [];
    const associatedMonitors = _.get(values, 'associatedMonitors', {});
    associatedMonitors.sequence.delegates.forEach((monitor) => {
      const filteredOption = monitors.filter((option) => option.monitor_id === monitor.monitor_id);
      monitorOptions.push({
        label: filteredOption[0]?.monitor_name || '',
        monitor_id: monitor.monitor_id,
      });
    });
    setOptions(monitorOptions);

    const condition = _.get(values, formikFullFieldName, '');

    let expressions = conditionToExpressions(condition, monitors);
    if (
      !edit &&
      !_.get(touched, formikFullFieldValue, false) &&
      triggerIndex === 0 &&
      expressions.length === 0
    ) {
      expressions = [];
      monitorOptions.forEach((monitor, index) => {
        expressions.push({
          description: index ? 'AND' : '',
          monitor_id: monitor.monitor_id,
          monitor_name: monitor.label,
        });
      });

      _.set(values, formikFullFieldName, expressionsToCondition(expressions));
    }

    setUsedExpressions(expressions?.length ? expressions : [DEFAULT_EXPRESSION]);
  };

  const expressionsToCondition = (expressions) => {
    const conditionMap = {
      AND: '&& ',
      OR: '|| ',
      NOT: '!',
      '': '',
      'AND NOT': '&& !',
      'OR NOT': '|| !',
    };

    const condition = expressions.reduce((query, expression) => {
      if (expression?.monitor_id) {
        query += ` ${conditionMap[expression.description]}monitor[id=${expression.monitor_id}]`;
        query = query.trim();
      }
      return query;
    }, '');

    return `(${condition})`;
  };

  const changeMonitor = (selection, exp, idx, form) => {
    const expressions = _.cloneDeep(usedExpressions);
    let monitor = selection[0];

    if (monitor?.monitor_id) {
      expressions[idx] = {
        ...expressions[idx],
        monitor_id: monitor.monitor_id,
        monitor_name: monitor.label,
      };
    } else {
      expressions[idx] = idx ? DEFAULT_NEXT_EXPRESSION : DEFAULT_EXPRESSION;
    }

    setUsedExpressions(expressions);
    onChange(form, expressions);
  };

  const changeCondition = (selection, exp, idx, form) => {
    const expressions = _.cloneDeep(usedExpressions);

    expressions[idx] = { ...expressions[idx], description: selection[0].description };
    setUsedExpressions(expressions);
    onChange(form, expressions);
  };

  const onChange = (form, expressions) => {
    form.setFieldValue(formikFullFieldName, expressionsToCondition(expressions));
  };

  const onBlur = (form, expressions) => {
    onChange(form, expressions);
    form.setFieldTouched(formikFullFieldValue, true);
  };

  const openPopover = (idx = 0) => {
    const expressions = _.cloneDeep(usedExpressions);
    expressions[idx] = { ...expressions[idx], isOpen: !expressions[idx].isOpen };
    setUsedExpressions(expressions);
  };

  const closePopover = (idx, form) => {
    const expressions = _.cloneDeep(usedExpressions);
    expressions[idx] = { ...expressions[idx], isOpen: false };
    setUsedExpressions(expressions);
    onBlur(form, expressions);
    form.setFieldTouched(`${expressionNamePrefix}_${idx}`, true);
  };

  const onRemoveExpression = useCallback(
    (form, idx) => {
      const expressions = _.cloneDeep(usedExpressions);
      expressions.splice(idx, 1);
      expressions.length && (expressions[0].description = '');

      if (!expressions?.length) {
        expressions.push(DEFAULT_EXPRESSION);
      }
      setUsedExpressions([...expressions]);
      onChange(form, expressions);
    },
    [usedExpressions]
  );

  const hasInvalidExpression = () =>
    !!usedExpressions.filter((expression) => expression.monitor_id === '')?.length;

  const isValid = () => options.length > 1 && usedExpressions.length > 1 && !hasInvalidExpression();

  const validate = () => {
    if (options.length < 2) return 'Trigger condition requires at least two associated monitors.';
    if (usedExpressions.length < 2)
      return 'Trigger condition requires at least two monitors selected.';
    if (hasInvalidExpression()) return 'Invalid expressions.';
  };

  const renderOptions = (expression, idx = 0, form) => (
    <EuiFlexGroup
      gutterSize="s"
      data-test-subj={`${formikFullFieldName}_${triggerIndex}_${idx}_options`}
    >
      <EuiFlexItem grow={false}>
        <EuiComboBox
          style={{ width: '150px' }}
          singleSelection={{ asPlainText: true }}
          data-test-subj={`condition-combobox-${triggerIndex}-${idx}`}
          compressed
          selectedOptions={[
            {
              label: expression.description,
              description: expression.description,
            },
          ]}
          onChange={(selection) => changeCondition(selection, expression, idx, form)}
          onBlur={() => onBlur(form, usedExpressions)}
          options={idx === 0 ? FIRST_EXPRESSION_CONDITIONS_MAP : EXPRESSION_CONDITIONS_MAP}
          autoFocus={false}
        />
      </EuiFlexItem>
      <EuiFlexItem grow={false}>{renderMonitorOptions(expression, idx, form)}</EuiFlexItem>
      <EuiFlexItem grow={false}>
        <EuiToolTip content={'Remove monitor'}>
          <EuiButtonIcon
            data-test-subj={`selection-exp-field-item-remove-${triggerIndex}-${idx}`}
            onClick={() => onRemoveExpression(form, idx)}
            iconType={'trash'}
            color="danger"
            aria-label={'Remove condition'}
            style={{ marginTop: '4px' }}
          />
        </EuiToolTip>
      </EuiFlexItem>
    </EuiFlexGroup>
  );

  const renderMonitorOptions = (expression, idx, form) => (
    <EuiComboBox
      singleSelection={{ asPlainText: true }}
      compressed
      onChange={(selection) => changeMonitor(selection, expression, idx, form)}
      onBlur={() => onBlur(form, usedExpressions)}
      selectedOptions={[
        {
          label: expression.monitor_name,
          monitor_id: expression.monitor_id,
        },
      ]}
      style={{ width: '250px' }}
      data-test-subj={`monitors-combobox-${triggerIndex}-${idx}`}
      options={(() => {
        const differences = _.differenceBy(options, usedExpressions, 'monitor_id');
        return [
          {
            monitor_id: expression.monitor_id,
            label: expression.monitor_name,
          },
          ...differences.map((sel) => ({
            monitor_id: sel.monitor_id,
            label: sel.label,
          })),
        ];
      })()}
    />
  );

  return (
    <FormikInputWrapper
      name={formikFullFieldValue}
      fieldProps={{
        validate: () => validate(),
      }}
      render={({ form }) => (
        <FormikFormRow
          name={formikFullFieldValue}
          form={form}
          rowProps={{
            isInvalid: () => form.touched[formikFullFieldValue] && !isValid(),
            error: () => validate(),
            style: {
              maxWidth: 'inherit',
            },
          }}
        >
          <EuiFlexGroup
            gutterSize="s"
            data-test-subj={formikFullFieldName}
            className={'expressionQueries'}
          >
            {usedExpressions.map((expression, idx) => (
              <EuiFlexItem grow={false} key={`selection_${triggerIndex}_${idx}`}>
                <EuiPopover
                  id={`selection_${idx}`}
                  button={
                    <EuiExpression
                      isInvalid={
                        form.touched[`${expressionNamePrefix}_${idx}`] && !expression.monitor_id
                      }
                      aria-label={'Add condition expression'}
                      description={expression.description}
                      value={expression.monitor_name}
                      isActive={expression.isOpen}
                      onClick={() => {
                        form.setFieldTouched(formikFullFieldValue, true);
                        openPopover(idx);
                      }}
                      data-test-subj={`select-expression_${triggerIndex}_${idx}`}
                    />
                  }
                  isOpen={expression.isOpen}
                  closePopover={() => closePopover(idx, form)}
                  panelPaddingSize="s"
                  anchorPosition="upCenter"
                >
                  {renderOptions(expression, idx, form)}
                </EuiPopover>
              </EuiFlexItem>
            ))}
            {options.length > usedExpressions.length && (
              <EuiFlexItem grow={false} key={`selection_add`}>
                <EuiToolTip content={'Add another monitor'}>
                  <EuiButtonIcon
                    onClick={() => {
                      const expressions = _.cloneDeep(usedExpressions);
                      expressions.push({
                        ...DEFAULT_NEXT_EXPRESSION,
                      });
                      setUsedExpressions(expressions);
                    }}
                    color={'primary'}
                    iconType="plusInCircleFilled"
                    aria-label={'Add one more condition'}
                    data-test-subj={`condition-add-options-btn_${triggerIndex}`}
                    style={{ marginTop: '1px' }}
                  />
                </EuiToolTip>
              </EuiFlexItem>
            )}
          </EuiFlexGroup>
        </FormikFormRow>
      )}
    />
  );
};

export default ExpressionBuilder;
