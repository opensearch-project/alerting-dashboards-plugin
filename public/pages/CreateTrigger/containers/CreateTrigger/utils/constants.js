/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const TRIGGER_TYPE = {
  AD: 'anomaly_detector_trigger',
  BUCKET_LEVEL: 'bucket_level_trigger',
  ALERT_TRIGGER: 'alerting_trigger',
  QUERY_LEVEL: 'query_level_trigger',
  DOC_LEVEL: 'document_level_trigger',
};

export const FORMIK_INITIAL_BUCKET_SELECTOR_VALUES = {
  buckets_path: {
    '': '',
  },
  parent_bucket_path: 'composite_agg',
  script: {
    source: 'params.',
  },
};

export const FORMIK_INITIAL_TRIGGER_CONDITION_VALUES = {
  thresholdValue: 10000,
  thresholdEnum: 'ABOVE',
  anomalyDetector: {
    triggerType: TRIGGER_TYPE.AD,
    anomalyGradeThresholdValue: 0.7,
    anomalyGradeThresholdEnum: 'ABOVE',
    anomalyConfidenceThresholdValue: 0.7,
    anomalyConfidenceThresholdEnum: 'ABOVE',
  },
  script: {
    lang: 'painless',
    source: 'ctx.results[0].hits.total.value > 0',
  },
  buckets_path: {},
  parent_bucket_path: 'composite_agg',
  gap_policy: '',
  query: undefined,
  queryMetric: undefined,
  andOrCondition: undefined,
};

export const FORMIK_INITIAL_TRIGGER_VALUES = {
  name: '',
  severity: '1',
  minTimeBetweenExecutions: null,
  rollingWindowSize: null,
  script: {
    lang: 'painless',
    source: `ctx.results[0].hits.total.value > 0`,
  },
  bucketSelector: JSON.stringify(FORMIK_INITIAL_BUCKET_SELECTOR_VALUES, null, 4), // TODO: To be used for Bucket-Level Triggers defined by query
  triggerConditions: [],
  thresholdValue: 10000,
  thresholdEnum: 'ABOVE',
  anomalyDetector: {
    triggerType: TRIGGER_TYPE.AD,
    anomalyGradeThresholdValue: 0.7,
    anomalyGradeThresholdEnum: 'ABOVE',
    anomalyConfidenceThresholdValue: 0.7,
    anomalyConfidenceThresholdEnum: 'ABOVE',
  },
  where: {
    fieldName: [],
    operator: 'includes',
    fieldValue: '',
    fieldRangeStart: 0,
    fieldRangeEnd: 0,
  },
  actions: undefined,
};

export const FORMIK_INITIAL_DOC_LEVEL_SCRIPT = {
  lang: FORMIK_INITIAL_TRIGGER_VALUES.script.lang,
  source: '(query[name=<queryName>] || query[name=<queryName>]) && query[tag=<queryTag>]',
};

export const HITS_TOTAL_RESULTS_PATH = 'ctx.results[0].hits.total.value';
export const AGGREGATION_RESULTS_PATH = 'ctx.results[0].aggregations.metric.value';
export const ANOMALY_GRADE_RESULT_PATH = 'ctx.results[0].aggregations.max_anomaly_grade.value';
export const ANOMALY_CONFIDENCE_RESULT_PATH = 'ctx.results[0].hits.hits[0]._source.confidence';
export const NOT_EMPTY_RESULT =
  'ctx.results != null && ctx.results.length > 0 && ctx.results[0].aggregations != null && ctx.results[0].aggregations.max_anomaly_grade != null && ctx.results[0].hits.total.value > 0 && ctx.results[0].hits.hits[0]._source != null && ctx.results[0].hits.hits[0]._source.confidence != null';
