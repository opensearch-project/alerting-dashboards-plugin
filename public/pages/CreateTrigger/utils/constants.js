/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import Message from '../components/Action/actions';

export const DEFAULT_MESSAGE_SOURCE = {
  BUCKET_LEVEL_MONITOR: `
  Monitor {{ctx.monitor.name}} just entered alert status. Please investigate the issue.
  - Trigger: {{ctx.trigger.name}}
  - Severity: {{ctx.trigger.severity}}
  - Period start: {{ctx.periodStart}}
  - Period end: {{ctx.periodEnd}}

  - Deduped Alerts:
  {{#ctx.dedupedAlerts}}
    * {{id}} : {{bucket_keys}}
  {{/ctx.dedupedAlerts}}

  - New Alerts:
  {{#ctx.newAlerts}}
    * {{id}} : {{bucket_keys}}
  {{/ctx.newAlerts}}

  - Completed Alerts:
  {{#ctx.completedAlerts}}
    * {{id}} : {{bucket_keys}}
  {{/ctx.completedAlerts}}
  `.trim(),
  QUERY_LEVEL_MONITOR: `
  Monitor {{ctx.monitor.name}} just entered alert status. Please investigate the issue.
  - Trigger: {{ctx.trigger.name}}
  - Severity: {{ctx.trigger.severity}}
  - Period start: {{ctx.periodStart}}
  - Period end: {{ctx.periodEnd}}
  `.trim(),
};

export const FORMIK_INITIAL_ACTION_VALUES = {
  name: '',
  destination_id: '',
  subject_template: {
    lang: 'mustache',
    source: 'Alerting Notification action',
  },
  message_template: {
    lang: 'mustache',
    source: DEFAULT_MESSAGE_SOURCE.QUERY_LEVEL_MONITOR,
  },
  throttle_enabled: false,
  throttle: {
    value: 10,
    unit: 'MINUTES', // throttle unit only supports MINUTES currently, no UI element mapped
  },
};

export const SEVERITY_OPTIONS = [
  { value: '1', text: '1 (Highest)' },
  { value: '2', text: '2 (High)' },
  { value: '3', text: '3 (Medium)' },
  { value: '4', text: '4 (Low)' },
  { value: '5', text: '5 (Lowest)' },
];

export const THRESHOLD_ENUM_OPTIONS = [
  { value: 'ABOVE', text: 'IS ABOVE' },
  { value: 'BELOW', text: 'IS BELOW' },
  { value: 'EXACTLY', text: 'IS EXACTLY' },
];

export const DEFAULT_AND_OR_CONDITION = 'AND';
export const AND_OR_CONDITION_OPTIONS = [
  { value: 'AND', text: 'AND' },
  { value: 'OR', text: 'OR' },
];

export const DEFAULT_TRIGGER_NAME = 'New trigger';
export const DEFAULT_ACTION_TYPE = 'slack';

export const MANAGE_CHANNELS_PATH = `/app/notifications-dashboards#/channels`;

export const webhookNotificationActionMessageComponent = (props) => (
  <Message isSubjectDisabled {...props} />
);
export const defaultNotificationActionMessageComponent = (props) => <Message {...props} />;
