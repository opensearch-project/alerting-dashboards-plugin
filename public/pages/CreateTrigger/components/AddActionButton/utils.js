/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import _ from 'lodash';
import { DEFAULT_MESSAGE_SOURCE, FORMIK_INITIAL_ACTION_VALUES } from '../../utils/constants';
import { MONITOR_TYPE } from '../../../../utils/constants';
import { getDigitId, getUniqueName } from '../../../../utils/helpers';

export const getInitialActionValues = ({ monitorType, flyoutMode, actions }) => {
  const initialActionValues = _.cloneDeep(FORMIK_INITIAL_ACTION_VALUES);

  switch (monitorType) {
    case MONITOR_TYPE.BUCKET_LEVEL:
      _.set(
        initialActionValues,
        'message_template.source',
        DEFAULT_MESSAGE_SOURCE.BUCKET_LEVEL_MONITOR
      );
      break;
    case MONITOR_TYPE.QUERY_LEVEL:
    case MONITOR_TYPE.CLUSTER_METRICS:
      _.set(
        initialActionValues,
        'message_template.source',
        DEFAULT_MESSAGE_SOURCE.QUERY_LEVEL_MONITOR
      );
      break;
  }

  const id = getDigitId();
  initialActionValues.id = `notification${id}`;

  // Add name based on previous name;
  if (flyoutMode) {
    initialActionValues.name = getUniqueName(actions, 'Notification ');
  }

  return initialActionValues;
};
