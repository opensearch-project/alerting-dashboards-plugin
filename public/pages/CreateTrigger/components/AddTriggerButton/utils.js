import _ from 'lodash';
import { getDigitId, getUniqueName } from '../../../../utils/helpers';
import { FORMIK_INITIAL_TRIGGER_VALUES } from '../../containers/CreateTrigger/utils/constants';
import { getInitialActionValues } from '../AddActionButton/utils';

export const getInitialTriggerValues = ({
  script = FORMIK_INITIAL_TRIGGER_VALUES.script,
  flyoutMode,
  triggers,
  monitorType,
}) => {
  const initialValues = _.cloneDeep({ ...FORMIK_INITIAL_TRIGGER_VALUES, script });
  const id = getDigitId();

  initialValues.id = `trigger${id}`;

  if (flyoutMode) {
    initialValues.name = getUniqueName(triggers, 'Trigger ');
    const initialAction = getInitialActionValues({ monitorType, flyoutMode, actions: [] });
    initialValues.actions = [initialAction];
  }

  return initialValues;
};
