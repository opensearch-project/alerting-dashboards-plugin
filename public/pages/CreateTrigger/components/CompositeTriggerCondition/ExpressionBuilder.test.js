/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';

import { Formik } from 'formik';
import { FORMIK_INITIAL_VALUES } from '../../../CreateMonitor/containers/CreateMonitor/utils/constants';
import ExpressionBuilder from './ExpressionBuilder';

describe('ExpressionBuilder', () => {
  test('renders', () => {
    const component = (
      <Formik initialValues={FORMIK_INITIAL_VALUES} onSubmit={() => {}}>
        <ExpressionBuilder
          formikFieldPath={'path'}
          formikFieldName={'triggerCondition'}
          values={{}}
          touched={{}}
          httpClient={{}}
        />
      </Formik>
    );
    expect(render(component)).toMatchSnapshot();
  });
});
