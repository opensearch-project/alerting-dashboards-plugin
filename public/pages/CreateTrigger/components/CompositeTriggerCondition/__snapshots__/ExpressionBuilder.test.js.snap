// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpressionBuilder renders 1`] = `
<div
  class="euiFormRow euiFormRow--compressed"
  id="pathtriggerCondition_value-form-row-row"
  style="max-width:inherit"
>
  <div
    class="euiFormRow__fieldWrapper"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiFlexGroup--responsive expressionQueries"
      data-test-subj="pathtriggerCondition"
      id="pathtriggerCondition_value-form-row"
    >
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiPopover euiPopover--anchorUpCenter"
          id="selection_0"
        >
          <div
            class="euiPopover__anchor"
          >
            <button
              aria-label="Add condition expression"
              class="euiExpression euiExpression-isClickable euiExpression-isUppercase euiExpression--success"
              data-test-subj="select-expression_undefined_0"
            >
              <span
                class="euiExpression__description"
              />
               
              <span
                class="euiExpression__value"
              >
                Select delegate monitor
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
